src/core/videosource.cpp

```diff
@@ -241,10 +242,19 @@ FFMS_VideoSource::FFMS_VideoSource(const char *SourceFile, FFMS_Index &Index, in
         // vc1 simply sets has_b_frames to 1 no matter how many there are so instead we set it to the max value
         // in order to not confuse our own delay guesses later
         // Doesn't affect actual vc1 reordering unlike h264
-        if (CodecContext->codec_id == AV_CODEC_ID_VC1 && CodecContext->has_b_frames)
+        if (CodecContext->codec_id == AV_CODEC_ID_VC1 && CodecContext->has_b_frames) {
             Delay = 7 + (CodecContext->thread_count - 1); // the maximum possible value for vc1
-        else
+        } else if (CodecContext->codec_id == AV_CODEC_ID_AV1) {
+#if VERSION_CHECK(LIBAVCODEC_VERSION_INT, >=, 60, 14, 101)
+            // libdav1d.c exports delay like this.
+            Delay = CodecContext->delay;
+#else
+            Delay = 0; // Welp, nothing we can do.
```

Comment:
Would it be better just to fail AV1 in this case? Seems like that might allow problems to be detected sooner than exporting corrupted frames.

Reply:
@myrsloik opinion?

Reply:
> @myrsloik opinion?
I'd simply print a warning or something. It's not worse than before. If we're lucky this process will drag long enough for FFmpeg 6.1 (I guess?) to be released and we could require it. Or did I misunderstand the problem?